src/basic/log.c

```diff
@@ -795,6 +795,7 @@ static void log_assert(
 
         log_abort_msg = buffer;
 
+        log_open();
```

Comment:
hmm, we should do log_open() in the case of assert() and assert_unreachable(), but not in the case of assert_return() really. (the former two are actually fatal and result in abort() the latter doesn't, and may be ignored). I think it's fine to open the log connection when we process a fatal error, but if we only process a transient error, then we probably should accept our fate, and leave things closed, since we might break stuff then, as returning with one more open fd might be a problem